### DISCUSSION TITLE
How to have a AUTO_INCREMENT field different from primary_key

### DISCUSSION THREAD

**Discussion by giovannism20** [fastapi/sqlmodel#1012] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
class ExampleTable(ExampleTableBase, table=True):
    __tablename__ = "example_table"

    id: UUID | None = Field(default_factory=uuid4, primary_key=True)
    another_auto_increment_field: str | None = Field()
```

### Description

Create a table `ExampleTable` with id as a primary key and other field called `another_auto_increment` as not a primary key.

### Operating System

Linux

### SQLModel Version

0.0.18

### Python Version

3.12

**Comment by jonyscathe** [comment]
For starters, probably don't make your field a string.
But only way I know of to do this is to use the `sa_column_kwargs` argument to set autoincrement=True so that will be passed through when SQLModel creates the sqlalchemy Column.
Not a very nice solution, but it works.

`another_auto_increment_field: int | None = Field(default=None, sa_column_kwargs={'autoincrement': True})`
